src/manifolds/SymplecticGrassmannStiefel.jl

```diff
@@ -0,0 +1,158 @@
+[`SymplecticStiefel`](@ref) manifolds tangent space at `p`.
+"""
+function check_vector(M::SymplecticGrassmann, p, X; kwargs...)
+    n, k = get_parameter(M.size)
+    return check_vector(SymplecticStiefel(2 * n, 2 * k), p, X; kwargs...)
+end
+
+embed(::SymplecticGrassmann, p) = p
+embed(::SymplecticGrassmann, p, X) = X
+embed!(::SymplecticGrassmann, q, p) = copyto!(q, p)
+embed!(::SymplecticGrassmann, Y, p, X) = copyto!(Y, X)
+embed!(::SymplecticGrassmann, q, p::StiefelPoint) = copyto!(q, p.value)
+embed!(::SymplecticGrassmann, Y, p::StiefelPoint, X::StiefelTVector) = copyto!(Y, X.value)
+embed(::SymplecticGrassmann, p::StiefelPoint) = p.value
+embed(::SymplecticGrassmann, p::StiefelPoint, X::StiefelTVector) = X.value
+
+@doc raw"""
+    exp(::SymplecticGrassmann, p, X)
+    exp!(M::SymplecticGrassmann, q, p, X)
+
+Compute the exponential mapping
+
+```math
+  \exp\colon T\mathrm{SpGr}(2n, 2k) → \mathrm{SpGr}(2n, 2k)
+```
+
+when representing points and tangent vectors as symplectic bases and their tangents, i.e.
+on the [`SymplecticStiefel`](@ref) manifold. Then we can just pass this on to [`exp(::SymplecticStiefel, p, X)`](@ref).
+"""
+exp(::SymplecticGrassmann, p, X)
+
+function exp!(M::SymplecticGrassmann, q, p, X)
+    n, k = get_parameter(M.size)
+    exp!(SymplecticStiefel(2 * n, 2 * k), q, p, X)
+    return q
+end
+
+function get_embedding(::SymplecticGrassmann{TypeParameter{Tuple{n,k}},𝔽}) where {n,k,𝔽}
+    return SymplecticStiefel(2n, 2k, 𝔽)
+end
+function get_embedding(M::SymplecticGrassmann{Tuple{Int,Int},𝔽}) where {𝔽}
+    n, k = get_parameter(M.size)
+    return SymplecticStiefel(2n, 2k, 𝔽; parameter=:field)
+end
+
+@doc raw"""
+    inner(::SymplecticGrassmann, p, X, Y)
+
+Compute the Riemannian inner product ``g^{\mathrm{SpGr}}_p(X,Y)``, where ``p``
+is a point on the [`SymplecticStiefel`](@ref) manifold and ``X,Y \in \mathrm{Hor}_p^π\operatorname{SpSt}(2n,2k)``
+are horizontal tangent vectors. The formula reads according to Proposition Lemma 4.8 [BendokatZimmermann:2021](@cite).
+
+```math
+g^{\mathrm{SpGr}}_p(X,Y) = \operatorname{tr}\bigl(
+        (p^{\mathrm{T}}p)^{-1}X^{\mathrm{T}}(I_{2n} - pp^+)Y
+    \bigr),
+```
+where ``I_{2n}`` denotes the identity matrix and ``(⋅)^+`` the [`symplectic_inverse`](@ref).
+"""
+function inner(M::SymplecticGrassmann, p, X, Y)
+    n, k = get_parameter(M.size)
+    J = SymplecticElement(p, X, Y) # in BZ21 also J
+    # Procompute lu(p'p) since we solve a^{-1}* 3 times
+    a = lu(p' * p) # note that p'p is symmetric, thus so is its inverse c=a^{-1}
+    # we split the original trace into two one with I -> (X'Yc)
+    # 1) we permute X' and Y c to c^{\mathrm{T}}Y^{\mathrm{T}}X = a\(Y'X) (avoids a large interims matrix)
+    # 2) the second we compute as c (X'p)(p^+Y) since both brackets are the smaller matrices
+    return tr(a \ (Y' * X)) - tr(
+        a \ ((X' * p) * symplectic_inverse_times(SymplecticStiefel(2 * n, 2 * k), p, Y)),
+    )
```

Comment:
Could you verify if this can be rewritten as:
```julia
return dot(a \ Y'), X) - dot(
        a \ (X' * p),  symplectic_inverse_times(SymplecticStiefel(2 * n, 2 * k), p, Y),
    )
```
(I've used tr(X*Y) = dot(X, Y) and associativity).

Reply:
I can check that carefully, sure.

Reply:
Besides a missing bracket in the beginning (hehe) this looks fine. I am currently running the tests for short.

Reply:
But to be honest, this formula took me also quite some time to transfer / understand / rewrite from the paper.

Reply:
OK, let's keep your version then.